wp-lazy-loading.php

```diff
@@ -118,57 +120,153 @@ function wp_lazy_loading_enabled( $tag_name, $context ) {
 }
 
 /**
- * Add `loading="lazy"` to `img` HTML tags.
+ * Filters specific tags in post content and modifies their markup.
  *
- * Currently the "loading" attribute is only supported for `img`, and is enabled by default.
+ * This function adds `srcset`, `sizes`, and `loading` attributes to `img` HTML tags.
  *
  * @since (TBD)
  *
+ * @see wp_image_tag_add_loading_attr()
+ * @see wp_image_tag_add_srcset_and_sizes_attr()
+ *
  * @param string $content The HTML content to be filtered.
  * @param string $context Optional. Additional context to pass to the filters. Defaults to `current_filter()` when not set.
- * @return string Converted content with 'loading' attributes added to images.
+ * @return string Converted content with images modified.
  */
-function wp_add_lazy_load_attributes( $content, $context = null ) {
+function wp_filter_content_tags( $content, $context = null ) {
 	if ( null === $context ) {
 		$context = current_filter();
 	}
 
-	if ( ! wp_lazy_loading_enabled( 'img', $context ) ) {
+	$add_loading_attr = wp_lazy_loading_enabled( 'img', $context );
+
+	if ( false === strpos( $content, '<img' ) ) {
 		return $content;
 	}
 
-	return preg_replace_callback(
-		'/<img\s[^>]+>/',
-		function( array $matches ) use( $content, $context ) {
-			if ( ! preg_match( '/\sloading\s*=/', $matches[0] ) ) {
-				$tag_html = $matches[0];
-
-				/**
-				 * Filters the `loading` attribute value. Default `lazy`.
-				 *
-				 * Returning `false` or an empty string will not add the attribute.
-				 * Returning `true` will add the default value.
-				 *
-				 * @since (TBD)
-				 *
-				 * @param string $default The filtered value, defaults to `lazy`.
-				 * @param string $tag_html The tag's HTML.
-				 * @param string $content The HTML containing the image tag.
-				 * @param string $context Optional. Additional context. Defaults to `current_filter()`.
-				 */
-				$value = apply_filters( 'wp_set_image_loading_attr', 'lazy', $tag_html, $content, $context );
+	if ( ! preg_match_all( '/<img\s[^>]+>/', $content, $matches ) ) {
+		return $content;
+	}
+
+	// List of the unique `img` tags found in $content.
+	$images = array();
 
-				if ( $value ) {
-					if ( ! in_array( $value, array( 'lazy', 'eager' ), true ) ) {
-						$value = 'lazy';
-					}
+	// List of the rest of the images, either local but without attachment_id, or remote.
+	$other_images = array();
```

Comment:
This is an unused remainder and should be removed :)

Reply:
Oops, missed it :)